src/codec.cairo

```diff
@@ -15,6 +15,15 @@ pub trait Encode<T> {
         Self::encode_to(self, ref dest);
         dest
     }
+
+    /// Encode using Bitcoin codec and return length byte size.
+    fn bytes_size(
+        self: @T
+    ) -> usize {
+        let mut dest: ByteArray = Default::default();
```

Comment:
Transactions happen to be very large. We don't want to encode them twice. `weight` calculation should be done along txid calculation, `validate_transaction` should return fee and txid to avoid duplication. @m-kus WDYT?

Reply:
I agree, let's then do all three things (txid, wtxid, and weight) in `fee_and_merkle_roots `? Maybe we will also be able to avoid the overlap between encode and encode_with_witness at some point

Reply:
> I agree, let's then do all three things (txid, wtxid, and weight) in `fee_and_merkle_roots `? Maybe we will also be able to avoid the overlap between encode and encode_with_witness at some point

Agree, Is much more straightforward. @Jeanmichel7 it is also an answer to your draft #132

Reply:
`fee_and_merkle_roots` function name should change.